### DISCUSSION TITLE
How the blazor get the render tree?

### DISCUSSION THREAD

**Discussion by ljzj2** [dotnet/aspnetcore#48890] [Q&A] [answered]
blazor webassembly we know it can update the dom after we change the variable.

I want to ask that how blazor get the render tree? will it get all the doms in the page?

we kow in react or vue or other framework it will get all the dom and build a list and compare which dom will update.

what about blazor?

**Comment by javiercn** [comment, marked as answer]
Blazor uses a Virtual DOM and produces a diff between the existing tree and the new tree for each component and then sends that diff to the browser which takes care of updating the DOM with the instructions from each component.
